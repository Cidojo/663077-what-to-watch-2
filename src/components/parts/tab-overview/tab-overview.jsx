import * as React from 'react';
import * as PropTypes from 'prop-types';

import {evalRatingLevel} from './../../../utils/utils.js';

const TabOverview = (props) => {
  const {
    description,
    director,
    starring,
    rating,
    scoresCount,
  } = props;

  return (
    <>
      <div className="movie-rating">
        <div className="movie-rating__score">{rating}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{evalRatingLevel(rating)}</span>
          <span className="movie-rating__count">{scoresCount} ratings</span>
        </p>
      </div>
      <div className="movie-card__text">
        <p>{description}</p>
        <p className="movie-card__director"><strong>Director: {director}</strong></p>

        <p className="movie-card__starring"><strong>Starring: {starring.join(`, `)} and
          other</strong></p>
      </div>
    </>
  );
};

TabOverview.propTypes = {
  rating: PropTypes.number,
  scoresCount: PropTypes.number,
  description: PropTypes.string,
  director: PropTypes.string,
  starring: PropTypes.arrayOf(PropTypes.string)
};

TabOverview.defaultProps = {
  rating: 0,
  scoresCount: 0,
  description: ``,
  director: ``,
  starring: []
};

export {TabOverview};
