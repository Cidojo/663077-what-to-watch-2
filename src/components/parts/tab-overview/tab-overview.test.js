import * as React from 'react';
import {TabOverview} from './tab-overview.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  rating: 0,
  scoresCount: 0,
  description: ``,
  director: ``,
  starring: []
};

it(`should render TabOverview component without errors`, () => {
  const tree = renderer.create(<TabOverview {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
