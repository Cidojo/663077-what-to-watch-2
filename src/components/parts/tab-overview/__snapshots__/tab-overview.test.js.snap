// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TabOverview component without errors 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      0
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        bad
      </span>
      <span
        className="movie-rating__count"
      >
        0
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        
      </strong>
    </p>
  </div>,
]
`;
